Alcancia-io/alcancia
# dashboard refactor

## Summary
Huge refactor:

- new folder structure 
- dashboard screen was refactored to achieve a better architecture
- several folders were included: services, queries, provider, models, queries, etc.
- Graphql widgets are no longer used
- User information is stored in provider
- dashboard screen widget was cleaned,  most of the logic is now handle in the controller


## Considerations
- There is an error when going to the user profile tab
- Other screens may be broken. I suggest we fix these things in other PRs

## Type of change

## How to test?
Login, wait for dashboard to load and check user info is correctly updated, go to user profile where username should show as well as the transactions, return to dashboard for user info to fetch again.

## How has this been tested?
- [ ] Toolbar should show username after dashboard loads
- [ ] User profit should be 0 and balance should load from backend
- [ ] User info should update when returning to dashboard
- [ ] Dashboard should transactions. If there aren't any, it should a corresponding message

## Checklist (for the reviewer)

- [ ] Is the PR free of merge conflicts?
- [ ] Is the UI responsive?
